File: using java/matrix/geeks for geeks/common_elements_in_all_rows_of_row-wise_sorted_matrix.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Codechef
{   
    public static void commonElements(int [][]matrix, int n, int m){
        Map<Integer,Integer> h = new HashMap<>();
        
        // Initialize the first row elements with value 1
        for(int j=0; j<m; j++){
            h.put(matrix[0][j],1);
        }
        
        // traverse the matrix except first row 
        for(int i=1; i<n; i++){
            for (int j=0; j<m; j++){
                
                // if element is present in the map and is not duplicated in the current row
                if(h.get(matrix[i][j])!=null && h.get(matrix[i][j]) == i){
                    h.put(matrix[i][j],i+1);
                    
                    // if this is last row
                    if(i==n-1){
                        System.out.print(matrix[i][j] + " ");
                    }
                }
            } 
        }
    }
    
	public static void main (String[] args) throws java.lang.Exception
	{
	    Scanner sc = new Scanner(System.in);
	    
	    int n = sc.nextInt();
	    int m = sc.nextInt();
	    
	    int [][]matrix = new int[n][m];
	    
	    for (int i=0; i<n; i++){
	        for(int j=0; j<m; j++){
	            matrix[i][j] = sc.nextInt();
	        }
	    } 
	    
	   commonElements(matrix, n, m); 
	}
}

File: using c++/matrix/geeks for geeks/common_elements_in_all_rows_of_row-wise_sorted_matrix.cpp
#include <bits/stdc++.h>
using namespace std;

void commonElements(vector<vector<int> > matrix, int n, int m){
        unordered_map<int,int> map;
        
        // Initialize the first row elements with value 1
        for(int j=0; j<m; j++){
            map[matrix[0][j]] = 1;
        }
        
        // traverse the matrix except first row 
        for(int i=1; i<n; i++){
            for (int j=0; j<m; j++){
                
                // if element is present in the map and is not duplicated in the current row
                if(map[matrix[i][j]] == i){
                    map[matrix[i][j]] = i+1;
                    
                    // if this is last row
                    if(i==n-1){
                        cout<<matrix[i][j]<<" ";
                    }
                }
            } 
        }
    }

int main() {
	int n,m;
        cin>>n>>m;
        vector<vector<int> > matrix(n); 

        for(int i=0; i<n; i++)
        {
            matrix[i].assign(m,0);
            for( int j=0; j<m; j++)
            {
                cin>>matrix[i][j];
            }
        }

   commonElements(matrix, n, m); 
   return 0;
}